enterprise/kernel/src/test/java/org/neo4j/kernel/impl/enterprise/lock/forseti/ForsetiFalseDeadlockTest.java

```diff
@@ -0,0 +1,388 @@
+        {
+            lockTypeBY.release( client, resourceType, 2 );
+        }
+
+        @Override
+        public String toString()
+        {
+            return "iterations=" + iterations +
+                   ", lockManager=" + lockManager +
+                   ", waitStrategy=" + waitStrategy +
+                   ", lockTypeAX=" + lockTypeAX +
+                   ", lockTypeAY=" + lockTypeAY +
+                   ", lockTypeBX=" + lockTypeBX +
+                   ", lockTypeBY=" + lockTypeBY;
+        }
+    }
+
+    public enum LockType
+    {
+        EXCLUSIVE
+                {
+                    @Override
+                    public void acquire( Locks.Client client, ResourceType resourceType, int resource )
+                    {
+                        client.acquireExclusive( resourceType, resource );
+                    }
+
+                    @Override
+                    public void release( Locks.Client client, ResourceType resourceType, int resource )
+                    {
+                        client.releaseExclusive( resourceType, resource );
+                    }
+                },
+        SHARED
+                {
+                    @Override
+                    public void acquire( Locks.Client client, ResourceType resourceType, int resource )
+                    {
+                        client.acquireShared( resourceType, resource );
+                    }
+
+                    @Override
+                    public void release( Locks.Client client, ResourceType resourceType, int resource )
+                    {
+                        client.releaseShared( resourceType, resource );
+                    }
+                };
+
+        public abstract void acquire( Locks.Client client, ResourceType resourceType, int resource );
+
+        public abstract void release( Locks.Client client, ResourceType resourceType, int resource );
+    }
+
+    public enum LockManager
+    {
+        COMMUNITY
+                {
+                    @Override
+                    public Locks create( ResourceType resourceType )
+                    {
+                        return new CommunityLockManger();
+                    }
+                },
+        FORSETI
+                {
+                    @Override
+                    public Locks create( ResourceType resourceType )
+                    {
+                        return new ForsetiLockManager( resourceType );
+                    }
+                };
+
+        public abstract Locks create( ResourceType resourceType );
+    }
+
+    @Parameterized.Parameters( name = "{0}" )
+    public static Iterable<Object[]> parameters()
+    {
+        List<Fixture> fixtures = new ArrayList<>();
+
+        // During development I also had iteration counts 1 and 2 here, but they never found anything, so for actually
+        // running this test, I leave only iteration count 100 enabled.
+        int[] iterations = new int[]{100};
```

Comment:
this singleton array is confusing - if not used, how about just inlining the value?